Tests/DatadogTests/Datadog/RUM/RUMVitals/VitalRefreshRateReaderTests.swift

```diff
@@ -11,43 +11,54 @@ import UIKit
 class VitalRefreshRateReaderTests: XCTestCase {
     private let mockNotificationCenter = NotificationCenter()
 
-    func testHighAndLowRefreshRates() {
+    func testWhenMainThreadOverheadGoesUp_itMeasuresLowerRefreshRate() throws {
```

Comment:
3 failures in last 1 month. This is quite old issue, I changed the approach for this test:
* Instead of observing main thread for certain amount of time...
* ... now it observes it until `VitalRefreshRateReader` records certain number of samples.

It passes 500x repetition in local. From my observation, time-based constraint was very flaky (sometimes recording just 1 sample, sometimes many). Current approach of recording 30 samples seems stable and still executes under `2s` in local (like before).